src/integer_mod_q/modulus_polynomial_ring_zq/get.rs

```diff
@@ -9,12 +9,56 @@
 //! Implementations to get content of a
 //! [`ModulusPolynomialRingZq].
 
-use super::ModulusPolynomialRingZq;
-use flint_sys::fq::fq_ctx_struct;
+use crate::integer_mod_q::{Modulus, ModulusPolynomialRingZq};
+use flint_sys::{fmpz_mod::fmpz_mod_ctx_init, fq::fq_ctx_struct};
+use std::{mem::MaybeUninit, rc::Rc};
 
 impl ModulusPolynomialRingZq {
     /// Returns the [`fq_ctx_struct`] of a modulus and is only used internally.
     pub(crate) fn get_fq_ctx_struct(&self) -> &fq_ctx_struct {
         self.modulus.as_ref()
     }
+
+    /// Returns a the prime q as a [`Modulus`]
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::{ModulusPolynomialRingZq, Modulus};
+    /// use std::str::FromStr;
+    ///
+    /// let modulus_ring = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    ///
+    /// let modulus = modulus_ring.get_q();
+    ///
+    /// # let cmp_modulus = Modulus::try_from(&17.into()).unwrap();
+    /// # assert_eq!(cmp_modulus, modulus);
+    /// ```
+    pub fn get_q(&self) -> Modulus {
```

Comment:
```suggestion
    /// Returns a the prime q as a [`Modulus`]
    ///
    /// # Examples
    /// ```
    /// use qfall_math::integer_mod_q::{ModulusPolynomialRingZq, Modulus};
    /// use std::str::FromStr;
    ///
    /// let modulus_ring = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
    ///
    /// let modulus = modulus_ring.get_mod();
    ///
    /// # let cmp_modulus = Modulus::try_from(&17.into()).unwrap();
    /// # assert_eq!(cmp_modulus, modulus);
    /// ```
    pub fn get_mod(&self) -> Modulus {
```

I think that the nameing is a off here.
I would expect `get_q` to return the q as a `Z` or so. Also, not consistent with `Zq` and `MatZq`.
Maybe call it `get_mod` or we make the fields of the `PolynomialRingZq` stuct public.